src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -155,7 +155,7 @@ private void EnableTimer(float pollingIntervalInSeconds)
             if (_pollingIntervalInMilliseconds == 0 || pollingIntervalInSeconds * 1000 < _pollingIntervalInMilliseconds)
             {
                 _pollingIntervalInMilliseconds = (int)(pollingIntervalInSeconds * 1000);
-                ResetCounters(); // Reset statistics for counters before we start the thread.
+                HandleCountersReset(); // Schedule IncrementingPollingCounter reset and synchronously reset other counters before we start the thread
```

Comment:
```suggestion
                // Schedule IncrementingPollingCounter reset and synchronously reset other counters
                HandleCountersReset(); 
```
Nit: The thread may already be running